content/guides/frameworks/laravel/development-setup.md

```diff
@@ -0,0 +1,342 @@
+      groupadd -g ${GID} www && \
+      useradd -m -u ${UID} -g www -s /bin/bash www; \
+    fi && \
+    usermod -aG sudo www && \
+    echo 'www ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers
+
+# Switch to the non-root user to install NVM and Node.js
+USER www
+
+# Install NVM (Node Version Manager) as the www user
+RUN export NVM_DIR="$HOME/.nvm" && \
+    curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.40.0/install.sh | bash && \
+    [ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh" && \
+    nvm install ${NODE_VERSION} && \
+    nvm alias default ${NODE_VERSION} && \
+    nvm use default
+
+# Ensure NVM is available for all future shells
+RUN echo 'export NVM_DIR="$HOME/.nvm"' >> /home/www/.bashrc && \
+    echo '[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"' >> /home/www/.bashrc && \
+    echo '[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"' >> /home/www/.bashrc
+
+# Set the working directory
+WORKDIR /var/www
+
+# Override the entrypoint to avoid the default php entrypoint
+ENTRYPOINT []
+
+# Default command to keep the container running
+CMD ["bash"]
+```
+
+### Create Docker Compose Configuration for Development
+
+Here's the `compose.yaml` file to set up the development environment:
+
+```yaml
+services:
+  web:
+    image: nginx:latest # Using the default Nginx image with custom configuration.
+    volumes:
+      # Mount the application code for live updates
+      - ./:/var/www
+      # Mount the Nginx configuration file
+      - ./docker/nginx/nginx.conf:/etc/nginx/nginx.conf:ro
+    ports:
+      # Map port 80 inside the container to the port specified by 'NGINX_PORT' on the host machine
+      - "${NGINX_PORT:-80}:80"
+    environment:
+      - NGINX_HOST=${NGINX_HOST}
+    networks:
+      - laravel
+    depends_on:
+      php-fpm:
+        condition: service_started  # Wait for php-fpm to start
+
+  php-fpm:
+    # For the php-fpm service, we will create a custom image to install the necessary PHP extensions and setup proper permissions.
+    build:
+      context: .
+      dockerfile: ./docker/php-fpm/Dockerfile
+      args:
+        UID: ${UID}
+        GID: ${GID}
+        XDEBUG_ENABLED: ${XDEBUG_ENABLED}
+        XDEBUG_MODE: ${XDEBUG_MODE}
+        XDEBUG_HOST: ${XDEBUG_HOST}
+        XDEBUG_IDE_KEY: ${XDEBUG_IDE_KEY}
+        XDEBUG_LOG: ${XDEBUG_LOG}
+        XDEBUG_LOG_LEVEL: ${XDEBUG_LOG_LEVEL}
+    env_file:
+      # Load the environment variables from the Laravel application
+      - .env
+    user: "${UID}:${GID}"
+    volumes:
+      # Mount the application code for live updates
+      - ./:/var/www
+    networks:
+      - laravel
+    depends_on:
+      postgres:
+        condition: service_started  # Wait for postgres to start
+
+  workspace:
+   # For the workspace service, we will also create a custom image to install and setup all the necessary stuff.
+    build:
+      context: .
+      dockerfile: ./docker/workspace/Dockerfile
+      args:
+        UID: ${UID}
+        GID: ${GID}
```

Comment:
If `GID` and `UID` env vars are unset, this would set `UID` and `GID` to empty strings, and thereby override the default `1000` set in the Dockerfile